src/test/rustdoc/check-source-code-urls-to-def-std.rs

```diff
@@ -15,3 +15,28 @@ pub fn foo(a: u32, b: &str, c: String) {
     let y: bool = true;
     babar();
 }
+
+macro_rules! yolo { () => {}}
+
+fn bar(a: i32) {}
+
+macro_rules! bar {
+    ($a:ident) => { bar($a) }
+}
+
+macro_rules! data{
+    ($x:expr) => { $x * 2 }
+}
+
+pub fn another_foo() {
+    // This is known limitation: if the macro doesn't generate anything, the visitor
+    // can't find any item or anything that could tell us that it comes from expansion.
+    // @!has - '//a[@href="../../src/foo/check-source-code-urls-to-def-std.rs.html#19"]' 'yolo!'
```

Comment:
A way to go around that limitation would be to `resolve` the macro (just like in intra-doc links). Might be better in a follow-up PR though.

Reply:
After taking a deeper look, it would require the `Resolver` from `DocContext`, so might be a bit bigger than what I originally imagined. I think doing in another PR makes even more sense.

Reply:
Please don't add more string-based resolving ... I haven't thought about this problem in detail, but looking it up by name alone doesn't seem like the right approach.

Reply:
Do you prefer a rustdoc-gui test instead?